### DISCUSSION TITLE
Row iteration

### DISCUSSION THREAD

**Discussion by mhmmd-iqbal** [rappasoft/laravel-livewire-tables#2019] [Q&A] [answered]
Hi, i need to show row iteration to my custom column No. 
ex. first row will show No. 1, second row will show 2, etc. 

How can i approach it?
![Screenshot from 2024-10-27 11-35-44](https://github.com/user-attachments/assets/914c2615-b154-4e43-ae68-4774be63951a)

**Comment by lrljoe** [comment, marked as answer]
So the simplest way to do this, assuming you want it to start at "1" not "0"

First you need to add an Alpine attribute to expose the Row's index, so you'll need to set the following:
```
public function configure(): void
{
    $this->setTrAttributes(function($row, int $index) {
        return [
            'default' => true,
            'x-data' => "{ rowIndex: ".($index + 1)." }",
        ];
    });
}
```

Then your Column can then make use of that "rowIndex" via an Alpine x-text:
```
Column::make("#")
    ->label(function ($row, $column) {
        return "<span x-text='rowIndex'></span>";
    })
    ->html(),
```

Note that there's no second parameter on this Column.  Setting a field/column that doesn't exist in your database, will result in an error.  

The first parameter in Column::make is the "label" for the Column, 
the second parameter in Column::make is the field/column/relation in your database (which isn't used for label columns)

**Reply by mhmmd-iqbal** [reply]
wow. It work. thankyou

**Reply by lrljoe** [reply]
No problem.  I will look at adding the row index to the tr by default early next month.

Once done, that would mean you wouldn't need to add the bit in configure()
